test/FastTests/Utils/TimeParsing.cs

```diff
@@ -151,7 +154,130 @@ public void TimeOnly(string date, int hh, int mm, int ss, int ms)
+        [InlineData("2024-12-13T02:38:42.7Z")] // 22
+        [InlineData("2024-12-13T02:38:42.77")] // 22
+        [InlineData("2024-12-13T02:38:42.7")] // 21
+        [InlineData("2024-12-13T02:38:42Z")]
+        public void CanParseValidDatesWithTrailingZerosInMillisecondsPart(string dt)
+        {
+            var expected = DateTime.ParseExact(dt, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture,
+                DateTimeStyles.RoundtripKind);
+
+            var bytes = Encoding.UTF8.GetBytes(dt);
+            fixed (byte* buffer = bytes)
+            {
+                Assert.Equal(LazyStringParser.Result.DateTime,
+                    LazyStringParser.TryParseDateTime(buffer, bytes.Length, out DateTime time, out _, properlyParseThreeDigitsMilliseconds: true));
 
+                Assert.Equal(expected.Kind, time.Kind);
+                Assert.Equal(expected, time);
+            }
+        }
 
+        [RavenTheory(RavenTestCategory.Core)]
+        [InlineData("2024-12-13T02:38:42.7864810Z")]
+        [InlineData("2024-12-13T02:38:42.7864800Z")]
+        [InlineData("2024-12-13T02:38:42.7864000Z")]
+        [InlineData("2024-12-13T02:38:42.7860000Z")]
+        [InlineData("2024-12-13T02:38:42.7800000Z")]
+        [InlineData("2024-12-13T02:38:42.7000000Z")]
+        [InlineData("2024-12-13T02:38:42.0000000Z")]
+        public void CanParseValidUtcDatesWithTrailingZerosInMillisecondsPart_DifferentFormats(string dt)
+        {
+            var expected = DateTime.ParseExact(dt, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture,
+                DateTimeStyles.RoundtripKind);
+
+            var formatsToRead = new Dictionary<string, DateTimeKind>
+            {
+                {DefaultFormat.DateTimeFormatsToRead[0], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[3], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[5], DateTimeKind.Utc},
+                {DefaultFormat.DateTimeFormatsToRead[6], DateTimeKind.Utc},
+            };
+
+            foreach (var dateTimeFormat in formatsToRead)
+            {
+                string tested = expected.ToString(dateTimeFormat.Key);
+
+                var expectedAfterFormatting = DateTime.ParseExact(tested, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture,
+                    DateTimeStyles.RoundtripKind);
+
+                var bytes = Encoding.UTF8.GetBytes(tested);
+                fixed (byte* buffer = bytes)
+                {
+                    Assert.Equal(LazyStringParser.Result.DateTime,
+                        LazyStringParser.TryParseDateTime(buffer, bytes.Length, out DateTime time, out _, properlyParseThreeDigitsMilliseconds: true));
+                    Assert.Equal(expectedAfterFormatting, time);
+                }
+            }
+        }
+
+        [RavenTheory(RavenTestCategory.Core)]
+        [InlineDataWithRandomSeed]
+        public void CanParseValidRandomDate(int seed)
```

Comment:
can we use that seed value as ticks in the `new DateTime(ticks, kind)` ctor? Thoughts?

Reply:
Not every random seed (passed as ticks) is a valid date .